src/runtime_src/core/common/query_requests.cpp

```diff
@@ -121,5 +122,48 @@ parse(const std::string& clock)
   auto clock_str = clock_map.find(clock);
   return clock_str != clock_map.end() ? clock_str->second : "N/A";
 }
+
+std::pair<uint64_t, uint64_t>
+xrt_core::query::xocl_errors::
+to_value(const std::vector<char>& buf, xrtErrorClass ecl)
+{
+  uint64_t errcode = 0;
+  uint64_t ts = 0;
+  if (buf.empty())
+    return {0, 0};
+  auto errors_buf = reinterpret_cast<const xcl_errors *>(buf.data());
+  if (errors_buf->num_err <= 0)
+    return {0, 0};
+  if (errors_buf->num_err > XCL_ERROR_CAPACITY)
+    throw xrt_core::system_error(EINVAL, "Invalid data in sysfs");
+
+  for (int i = errors_buf->num_err; i >= 0; i--) {
+    if (XRT_ERROR_CLASS(errors_buf->errors[i].err_code) != ecl)
+      continue;
+    errcode = errors_buf->errors[i].err_code;
+    ts = errors_buf->errors[i].ts;
+    break;
```

Comment:
Maybe just
```
if (XRT_ERROR_CLASS(errors_buf->errors[i].err_code) != ecl)
      return {errors_buf->errors[i].err_code, errors_buf->errors[i].ts};
```

And remove line 130, 131

Reply:
updated